api/client/ffs.go

```diff
@@ -236,6 +268,48 @@ func (f *ffs) Get(ctx context.Context, c cid.Cid) (io.Reader, error) {
 	return reader, nil
 }
 
+func (f *ffs) WatchLogs(ctx context.Context, ch chan<- LogEvent, c cid.Cid, opts ...WatchLogsOption) error {
+	config := WatchLogsConfig{}
+	for _, opt := range opts {
+		opt(&config)
+	}
+
+	stream, err := f.client.WatchLogs(ctx, &rpc.WatchLogsRequest{Cid: c.String(), Jid: config.jid.String()})
+	if err != nil {
+		return err
+	}
+	go func() {
+		for {
+			reply, err := stream.Recv()
+			if err == io.EOF || status.Code(err) == codes.Canceled {
```

Comment:
Ok @jsign I added another error condition where you just want to close the channel and return. In this case when the user cancels the context they passed into this `WatchLogs` method, that cancels the server side process as well and returns an error of `context canceled`. In that case, I don't want to send a `LogEvent` containing the error because it is expected behavior. Made the same change in `WatchJobs` above.

Reply:
Sounds good to me.